tripper/mappings/mappings.py

```diff
@@ -0,0 +1,640 @@
+    # pylint: disable=too-few-public-methods
+
+    def __init__(
+        self,
+        value: "Any",
+        unit: "Optional[str]" = None,
+        iri: "Optional[str]" = None,
+        property_iri: "Optional[str]" = None,
+        cost: float = 0.0,
+    ):
+        self.value = value
+        self.unit = unit
+        self.iri = iri
+        self.property_iri = property_iri
+        self.cost = cost
+
+    def show(
+        self,
+        routeno: "Optional[int]" = None,
+        name: "Optional[str]" = None,
+        indent: int = 0,
+    ):  # pylint: disable=unused-argument
+        """Returns a string representation of the Value.
+
+        Arguments:
+            routeno: Unused.  The argument exists for consistency with
+                the corresponding method in Step.
+            name: Name of value.
+            indent: Indentation level.
+        """
+        strings = []
+        ind = " " * indent
+        strings.append(ind + f'{name if name else "Value"}:')
+        strings.append(ind + f"  iri: {self.iri}")
+        strings.append(ind + f"  property_iri: {self.property_iri}")
+        strings.append(ind + f"  unit: {self.unit}")
+        strings.append(ind + f"  cost: {self.cost}")
+        strings.append(ind + f"  value: {self.value}")
+        return "\n".join(strings)
+
+
+class MappingStep:
+    """A step in a mapping route from a target to one or more sources.
+
+    A mapping step corresponds to one or more RDF triples.  In the
+    simple case of a `mo:mapsTo` or `rdfs:isSubclassOf` relation, it is
+    only one triple.  For transformations that has several input and
+    output, a set of triples are expected.
+
+    Arguments:
+        output_iri: IRI of the output concept.
+        steptype: One of the step types from the StepType enum.
+        function: Callable that evaluates the output from the input.
+        cost: The cost related to this mapping step.  Should be either a
+            float or a callable taking the same arguments as `function` as
+            input returning the cost as a float.
+        output_unit: Output unit.
+
+    The arguments can also be assigned as attributes.
+    """
+
+    # pylint: disable=too-many-instance-attributes
+
+    def __init__(
+        self,
+        output_iri: "Optional[str]" = None,
+        steptype: "Optional[StepType]" = None,
+        function: "Optional[Callable]" = None,
+        cost: "Union[float, Callable]" = 1.0,
+        output_unit: "Optional[str]" = None,
+    ):
+        self.output_iri = output_iri
+        self.steptype = steptype
+        self.function = function
+        self.cost = cost
+        self.output_unit = output_unit
+        self.input_routes: "List[dict]" = []  # list of inputs dicts
+        self.join_mode = False  # whether to join upcoming input
+        self.joined_input: "Inputs" = {}
+
+    def add_inputs(self, inputs: "Inputs"):
+        """Add input dict for an input route."""
+        assert isinstance(inputs, dict)  # nosec
```

Comment:
```suggestion
        assert isinstance(inputs, dict)  # nosec B101
```
To specify what test we are skipping.